Maple2.Server.Game/Manager/HousingManager.cs

```diff
@@ -297,16 +318,230 @@ public void InitNewHome(string characterName, ExportedUgcMapMetadata? template)
+
+        result.Position = position;
+        result.Rotation = rotation;
+        plot.Cubes.Add(position, result);
+        return true;
+    }
+
+    public bool TryRemoveCube(Plot plot, in Vector3B position, [NotNullWhen(true)] out PlotCube? cube) {
+        if (!plot.Cubes.Remove(position, out cube)) {
+            session.Send(CubePacket.Error(UgcMapError.s_ugcmap_no_cube_to_remove));
+            return false;
+        }
+
+        if (plot.IsPlanner) {
+            return true;
+        }
+
+        if (!session.Item.Furnishing.RetrieveCube(cube.Id)) {
+            throw new InvalidOperationException($"Failed to deposit cube {cube.Id} back into storage.");
+        }
+
+        return true;
+    }
+
+    public bool IsCoordOutsideArea(Vector3B position) {
+        int height = Home.IsPlanner ? Home.PlannerHeight : Home.Height;
+        int area = Home.IsPlanner ? Home.PlannerArea : Home.Area;
+
+        // Check if the position is outside the planar area bounds
+        if (position.X > 0 || position.Y > 0 || position.Z < 0) {
+            return true;
+        }
+
+        area *= -1;
+        if (position.X <= area || position.Y <= area) {
+            return true;
+        }
+
+        // Check if the position is outside the height bounds
+        if (position.Z > height) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public void RequestLayout(HomeLayout layout) {
+        Dictionary<int, int> groupedCubes = layout.Cubes.GroupBy(plotCube => plotCube.ItemId)
+            .ToDictionary(grouping => grouping.Key, grouping => grouping.Count()); // Dictionary<item id, count>
+        int cubeCount = 0;
+        Dictionary<FurnishingCurrencyType, long> cubeCosts = new() {
+            { FurnishingCurrencyType.Meso, 0 },
+            { FurnishingCurrencyType.Meret, 0 },
+        };
+
+        foreach ((int id, int amount) in groupedCubes) {
+            tableMetadata.FurnishingShopTable.Entries.TryGetValue(id, out FurnishingShopTable.Entry? shopEntry);
+            if (shopEntry is null) {
+                Log.Logger.Error("Failed to get shop entry for cube {cubeId}.", id);
+                session.Send(CubePacket.Error(UgcMapError.s_err_cannot_buy_limited_item_more));
+                return;
+            }
+
+            Item? item = session.Item.Furnishing.GetItem(id);
+            if (item is null) {
+                cubeCosts[shopEntry.FurnishingTokenType] += shopEntry.Price * amount;
+                cubeCount += amount;
+                continue;
+            }
+
+            if (item.Amount >= amount) {
+                continue;
+            }
+
+            int missingCubes = amount - item.Amount;
+            cubeCosts[shopEntry.FurnishingTokenType] += shopEntry.Price * missingCubes;
+            cubeCount += missingCubes;
+        }
+
+        session.Send(CubePacket.BuyCubes(cubeCosts, cubeCount));
+    }
+
+    public void ApplyLayout(Plot plot, HomeLayout layout, bool isBlueprint = false) {
```

Comment:
**suggestion:** Reconsider the necessity of the isBlueprint parameter

The isBlueprint parameter in ApplyLayout is only used in one place. Consider if this parameter is necessary or if there's a more elegant way to handle this logic, perhaps by creating separate methods for blueprints and regular layouts.